### PR TITLE
Add registry option to the Bootable Volume create form

### PR DESCRIPTION
**Pull request by upalatucci**
## 📝 Description

DOC: https://docs.google.com/document/d/1oETHC4UpRnEGFEdiezUxMLzpQagWXOYz0DBYro-XHB0/edit

Merge bootable volume  and DS modals

- add the registry DS option

The `automatic update` checkbox that is in the design needs to be addressed a little bit more, but if necessary can be added afterward

## 🎥 Demo

**Before**

image

**After**

image

image

### CODE REVIEW

File: src/utils/components/AddBootableVolumeModal/components/SourceTypeSelection/SourceTypeSelection.tsx [hunk 1]
```diff
@@ -0,0 +1,56 @@
...
+
+import { DROPDOWN_FORM_SELECTION } from '../../utils/constants';
+
+type SourceTypeSelectionProps = {
+  formSelection: DROPDOWN_FORM_SELECTION;
+  setFormSelection: React.Dispatch<React.SetStateAction<DROPDOWN_FORM_SELECTION>>;
+};
+
+const SourceTypeSelection: FC<SourceTypeSelectionProps> = ({ formSelection, setFormSelection }) => {
+  const [isOpen, setIsOpen] = React.useState(false);
```
Comment (metalice) [inline src/utils/components/AddBootableVolumeModal/components/SourceTypeSelection/SourceTypeSelection.tsx:14]: can remove react

Comment (upalatucci) [reply src/utils/components/AddBootableVolumeModal/components/SourceTypeSelection/SourceTypeSelection.tsx]: removed

File: src/views/catalog/CreateFromInstanceTypes/components/AddBootableVolumeButton/AddBootableVolumeButton.tsx [hunk 2]
```diff
@@ -17,6 +17,9 @@ export type AddBootableVolumeButtonProps = {
 const AddBootableVolumeButton: FC<AddBootableVolumeButtonProps> = ({ buttonVariant }) => {
   const { t } = useKubevirtTranslation();
   const { createModal } = useModal();
+
+  const sourceNamespace = isUpstream ? KUBEVIRT_OS_IMAGES_NS : OPENSHIFT_OS_IMAGES_NS;
+
   const [canCreateVolume] = useAccessReview({
     group: DataSourceModel.apiGroup,
     namespace: isUpstream ? KUBEVIRT_OS_IMAGES_NS : OPENSHIFT_OS_IMAGES_NS,
```
Comment (metalice) [inline src/views/catalog/CreateFromInstanceTypes/components/AddBootableVolumeButton/AddBootableVolumeButton.tsx:25]: line 21 is the same?

File: src/utils/components/AddBootableVolumeModal/utils/utils.ts [hunk 3]
```diff
@@ -29,23 +43,32 @@ export const createBootableVolume =
       storageClassName,
       uploadFile,
     } = bootableVolume || {};
+
+    let newDataSource: V1beta1DataSource;
```
Comment (metalice) [inline src/utils/components/AddBootableVolumeModal/utils/utils.ts:47]: no need for a let here , there is 2 places u are using it, one is scoped, u can declare it at the scope level at line 57 and also in line 97

File: src/utils/components/AddBootableVolumeModal/utils/utils.ts [hunk 4]
```diff
@@ -71,10 +87,87 @@ export const createBootableVolume =
...
+    draft.spec = {
+      garbageCollect: 'Outdated',
+      importsToKeep: retainRevisions,
+      managedDataSource: bootableVolumeName,
+      schedule: cronExpression,
+      template: {
+        spec: {
+          source: {
+            registry: {
+              url: registryURL.includes('docker://') ? registryURL : 'docker://' + registryURL,
```
Comment (metalice) [inline src/utils/components/AddBootableVolumeModal/utils/utils.ts:131]: i think we have functions of dockerPrefix or something like that, please check

File: src/utils/components/AddBootableVolumeModal/utils/utils.ts (hunk 3 above)
Comment (upalatucci) [reply src/utils/components/AddBootableVolumeModal/utils/utils.ts:47]: thanks

File: src/utils/components/AddBootableVolumeModal/utils/utils.ts (hunk 4 above)
Comment (upalatucci) [reply src/utils/components/AddBootableVolumeModal/utils/utils.ts:131]: Okay. The function is called `appendDockerPrefix` and it's defined in the catalog. 
After this pr I'll create another pr to move this function in the utils folder.

File: src/views/catalog/CreateFromInstanceTypes/components/AddBootableVolumeButton/AddBootableVolumeButton.tsx (hunk 2 above)
Comment (upalatucci) [reply src/views/catalog/CreateFromInstanceTypes/components/AddBootableVolumeButton/AddBootableVolumeButton.tsx]: Ah yeah. I solved a conflict there. Fixed

File: src/utils/components/AddBootableVolumeModal/components/SourceTypeSelection/SourceTypeSelection.tsx [hunk 5]
```diff
@@ -0,0 +1,56 @@
...
+type SourceTypeSelectionProps = {
+  formSelection: DROPDOWN_FORM_SELECTION;
+  setFormSelection: Dispatch<SetStateAction<DROPDOWN_FORM_SELECTION>>;
+};
+
+const SourceTypeSelection: FC<SourceTypeSelectionProps> = ({ formSelection, setFormSelection }) => {
+  const [isOpen, setIsOpen] = useState(false);
+
+  const onSelect = useCallback(
+    (event, value) => {
```
[the remark this answers is not in the document]
Comment (upalatucci) [reply src/utils/components/AddBootableVolumeModal/components/SourceTypeSelection/SourceTypeSelection.tsx:17]: still works yeah

**Pull request merged**